internal/serve/serve.go

```diff
@@ -296,9 +296,14 @@ func handleHTTP(o ServeOptions) *chi.Mux {
 
 		r.With(middleware.AnyRoleMiddleware(authManager, data.GetAllRoles()...)).Route("/wallets", func(r chi.Router) {
 			walletsHandler := httphandler.WalletsHandler{Models: o.Models}
+
 			r.Get("/", walletsHandler.GetWallets)
+
 			r.With(middleware.AnyRoleMiddleware(authManager, data.OwnerUserRole)).
 				Post("/", walletsHandler.PostWallets)
+
+			r.With(middleware.AnyRoleMiddleware(authManager, data.OwnerUserRole)).
+				Post("/{id}", walletsHandler.DeleteWallet)
```

Comment:
Please don't forget to add this endpoint to the Test_handleHTTP_authenticatedEndpoints test:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/1fe89478630c1c3f8fe2d4ed8c51227a010bbeef/internal/serve/serve_test.go#L255-L278